client/src/components/UploadComponent.js

```diff
@@ -0,0 +1,339 @@
+import React, { useState, useRef } from "react";
```

Comment:
No need to import React
```suggestion
import { useState, useRef } from "react";
```